dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -1,387 +1,388 @@
+                    outcome_train = self._noise(outcome_train, **func_args)
+                outcome_validation = self._noise(outcome_validation, **func_args)
+            elif action == 'permute':
+                if X_train is not None:
+                    outcome_train = self._permute(outcome_train, **func_args)
+                outcome_validation = self._permute(outcome_validation, **func_args)
+            elif action =='zero':
+                if X_train is not None:
+                    outcome_train = np.zeros(outcome_train.shape)
+                outcome_validation = np.zeros(outcome_validation.shape)
+
+        return outcome_validation
+            
+    def check_for_estimator(self):
+        """
+        This function checks if there is an estimator in the transformation list.
+        If there are no estimators, it allows us to optimize processing by skipping the 
+        data preprocessing and running the transformations on the whole dataset.
+        """
+        for action,_ in self._transformation_list:
+            if callable(action) or action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
+                return False
+            
+        return True
+
+    def preprocess_data_by_treatment(self):
+        """
+        This function groups data based on the data type of the treatment.
+        
+        Expected variable types supported for the treatment:
+        * bool
+        * pd.categorical
+        * float
+        * int
+        
+        returns pandas.core.groupby.generic.DataFrameGroupBy
+        """
+        assert len(self._treatment_name) == 1, "At present, DoWhy supports a simgle treatment variable"
+        
+        treatment_variable_name = self._target_estimand.treatment_name[0] # As we only have a single treatment
+        variable_type = self._data[treatment_variable_name].dtypes
+        
+        if bool == variable_type:
+            groups = self._data.groupby(treatment_variable_name)
+            return groups
+        # We use string arguments to account for both 32 and 64 bit varaibles
+        elif 'float' in variable_type.name or \
+               'int' in variable_type.name:
+            # action for continuous variables
+            data =  self._data
+            std_dev = data[treatment_variable_name].std()
+            num_bins = ( data.max() - data.min() )/ (self._bucket_size_scale_factor * std_dev)
+            data['bins'] = pd.cut(data[treatment_variable_name], num_bins)
+            groups = data.groupby('bins')
+            data.drop('bins')
+            return groups
+
+        elif 'categorical' in variable_type.name:
+            # Action for categorical variables
+            groups = data.groupby(treatment_variable_name)
+            groups = data.groupby('bins')
+            return groups
+        else:
+            raise ValueError("Passed {}. Expected bool, float, int or categorical.".format(variable_type.name))
+
+        # return data_chunks
```

Comment:
This line can be removed. We do not need ```data_chunks``` any longer

Reply:
sure.